californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -252,10 +256,12 @@ public CoapClient setDestinationContext(EndpointContext peerContext) {
 	 */
 	public CoapClient useExecutor() {
 		boolean failed = true;
-		ExecutorService executor = ExecutorsUtil.newFixedThreadPool(1, new NamedThreadFactory("CoapClient#")); //$NON-NLS-1$
+		ExecutorService executor = ExecutorsUtil.newFixedThreadPool(1, new NamedThreadFactory("CoapClient(main)#")); //$NON-NLS-1$
+		ScheduledThreadPoolExecutor secondaryExecutor = new ScheduledThreadPoolExecutor(1, new NamedThreadFactory("CoapClient(secondary)#"));
```

Comment:
```
if (failed) {
	executor.shutdownNow();
	throw new IllegalStateException("Executor already set or used!");
}
```

Should that be extended for the `secondaryExecutor` ?

Reply:
any reason not to use the `ExecutorsUtil.newDefaultSecondaryScheduler`?

Reply:
yep